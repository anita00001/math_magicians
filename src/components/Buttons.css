button {
  height: 2.25rem;
  border: 0.03em solid rgb(215, 195, 195);
  font-size: 0.5em;
}

.zero {
  width: 50%;
}

.oper-right {
  width: 25%;
  background-color: rgba(245, 124, 38, 0.884);
}

.calc-panel {
  width: 25%;
}
