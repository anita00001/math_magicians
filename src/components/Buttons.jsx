// Buttons.js
import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Buttons.css';

function Buttons(props) {
  const { labels, onButtonClick } = props;

  const handleClick = (label) => {
    onButtonClick(label);
  };

  return (
    <div className="button-container">
      {labels.map((label) => {
        const uniqueKey = `button-${label}`;
        let buttonClass = '';
        if (label === '0') {
          buttonClass = 'zero';
        } else if (label === 'รท') {
          buttonClass = 'oper-right';
        } else if (label === 'x') {
          buttonClass = 'oper-right';
        } else if (label === '-') {
          buttonClass = 'oper-right';
        } else if (label === '+') {
          buttonClass = 'oper-right';
        } else if (label === '=') {
          buttonClass = 'oper-right';
        } else {
          buttonClass = 'calc-panel';
        }
        return (
          <button
            key={uniqueKey}
            type="button"
            className={buttonClass}
            onClick={() => handleClick(label)}
          >
            {label}
          </button>
        );
      })}
    </div>
  );
}

Buttons.propTypes = {
  labels: PropTypes.arrayOf(PropTypes.string).isRequired,
  onButtonClick: PropTypes.func.isRequired,
};

export default Buttons;
