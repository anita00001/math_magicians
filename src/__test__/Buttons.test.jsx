import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Buttons from '../components/Buttons';

describe('Buttons', () => {
  const labels = [
    'AC', '+/-', '%', 'รท',
    '7', '8', '9', 'x',
    '4', '5', '6', '-',
    '1', '2', '3', '+',
    '0', '.', '=',
  ];
  const mockButtonClick = jest.fn();

  test('renders correctly', () => {
    const { container } = render(
      <Buttons labels={labels} onButtonClick={mockButtonClick} />,
    );
    expect(container).toMatchSnapshot();
  });

  test('Call onButtonClick correctly when a button is clicked', () => {
    const { getByText } = render(
      <Buttons labels={labels} onButtonClick={mockButtonClick} />,
    );
    fireEvent.click(getByText('1'));
    expect(mockButtonClick).toHaveBeenCalledWith('1');
    expect(mockButtonClick).toHaveBeenCalledTimes(1);

    fireEvent.click(getByText('='));
    expect(mockButtonClick).toHaveBeenCalledWith('=');
    expect(mockButtonClick).toHaveBeenCalledTimes(2);
  });
});
