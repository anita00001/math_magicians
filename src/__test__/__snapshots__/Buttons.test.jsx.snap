// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons renders correctly 1`] = `
<div>
  <div
    class="button-container"
  >
    <button
      class="calc-panel"
      type="button"
    >
      AC
    </button>
    <button
      class="calc-panel"
      type="button"
    >
      +/-
    </button>
    <button
      class="calc-panel"
      type="button"
    >
      %
    </button>
    <button
      class="oper-right"
      type="button"
    >
      รท
    </button>
    <button
      class="calc-panel"
      type="button"
    >
      7
    </button>
    <button
      class="calc-panel"
      type="button"
    >
      8
    </button>
    <button
      class="calc-panel"
      type="button"
    >
      9
    </button>
    <button
      class="oper-right"
      type="button"
    >
      x
    </button>
    <button
      class="calc-panel"
      type="button"
    >
      4
    </button>
    <button
      class="calc-panel"
      type="button"
    >
      5
    </button>
    <button
      class="calc-panel"
      type="button"
    >
      6
    </button>
    <button
      class="oper-right"
      type="button"
    >
      -
    </button>
    <button
      class="calc-panel"
      type="button"
    >
      1
    </button>
    <button
      class="calc-panel"
      type="button"
    >
      2
    </button>
    <button
      class="calc-panel"
      type="button"
    >
      3
    </button>
    <button
      class="oper-right"
      type="button"
    >
      +
    </button>
    <button
      class="zero"
      type="button"
    >
      0
    </button>
    <button
      class="calc-panel"
      type="button"
    >
      .
    </button>
    <button
      class="oper-right"
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
